### PR TITLE
Add hardware encoding status to playback data

### LINKED ISSUE #6087: Playback data: add info on which encoding (HW or SF) is used
Currently it's a bit hard to troubleshoot if HW or SW encoding is being used - need to dive into Jellyfin transcoding logs, or sometimes even check on the container  / OS if the right package is installed.

Would be nice to see at-a-glance if Hardware encoding is working as intended or not. Would there be a way to add this information to Settings > Playback Data under "Transcoding info" ?

### PR DESCRIPTION
**Pull request by gnuyent**
Hi, this is my first contribution to jellyfin. Let me know if I need to change/fix anything!

**Changes**
Inside Playback Info, add hardware encoding information.
image
image

**Issues**

Resolves https://github.com/jellyfin/jellyfin/issues/6087

Connected PR: https://github.com/jellyfin/jellyfin-web/pull/2776

### CODE REVIEW

File: Jellyfin.Api/Helpers/TranscodingJobHelper.cs [hunk 1]
```diff
@@ -458,6 +458,7 @@ private void DeleteHlsPartialStreamFiles(string outputFilePath)
                     AudioChannels = state.OutputAudioChannels,
                     IsAudioDirect = EncodingHelper.IsCopyCodec(state.OutputAudioCodec),
                     IsVideoDirect = EncodingHelper.IsCopyCodec(state.OutputVideoCodec),
+                    IsHardwareEncode = !string.IsNullOrEmpty(_serverConfigurationManager.GetEncodingOptions().HardwareAccelerationType),
```
Comment (MrTimscampi) [inline Jellyfin.Api/Helpers/TranscodingJobHelper.cs:461]:
```suggestion
                    HardwareAccelerationType = _serverConfigurationManager.GetEncodingOptions().HardwareAccelerationType,
```

File: MediaBrowser.Model/Session/TranscodingInfo.cs [hunk 2]
```diff
@@ -34,6 +34,8 @@ public TranscodingInfo()
 
         public int? AudioChannels { get; set; }
 
+        public bool IsHardwareEncode { get; set; }
```
Comment (MrTimscampi) [inline MediaBrowser.Model/Session/TranscodingInfo.cs:37]:
```suggestion
        public string HardwareAccelerationType { get; set; }
```

**Review by MrTimscampi — COMMENTED** [review]
From a client developer perspective, the proposed changes would be more useful than a simple boolean, and the nomenclature mirroring that of the server makes it easier to go look into the server code if we have issues or questions about what it does and how it works.

From a user perspective, it also gives them more information as to what is going on.

File: Jellyfin.Api/Helpers/TranscodingJobHelper.cs (hunk 1 above)
Comment (gnuyent) [reply Jellyfin.Api/Helpers/TranscodingJobHelper.cs]: Done. Added a helper variable to make the line shorter and to handle cases where hardware acceleration is not used.

File: Jellyfin.Api/Helpers/TranscodingJobHelper.cs [hunk 3]
```diff
@@ -444,6 +444,8 @@ private void DeleteHlsPartialStreamFiles(string outputFilePath)
             {
                 var audioCodec = state.ActualOutputAudioCodec;
                 var videoCodec = state.ActualOutputVideoCodec;
+                var hardwareAccelerationType = _serverConfigurationManager.GetEncodingOptions().HardwareAccelerationType;
+                hardwareAccelerationType = string.IsNullOrEmpty(hardwareAccelerationType) ? "none" : hardwareAccelerationType;
```
Comment (MrTimscampi) [inline Jellyfin.Api/Helpers/TranscodingJobHelper.cs:448]: I'm not a server dev, but usually the API simply returns `null` when something isn't set, which is likely better than returning a string here.

Do wait for someone from the server team to confirm this, though.

Comment (nielsvanvelzen) [reply Jellyfin.Api/Helpers/TranscodingJobHelper.cs:448]: Additionally using an enum is preferred so the client devs know the possible values.

Comment (crobibero) [reply Jellyfin.Api/Helpers/TranscodingJobHelper.cs:448]: I concur, enum if possible, null if not set.
I would give more input but currently on vacation without laptop

Comment (gnuyent) [reply Jellyfin.Api/Helpers/TranscodingJobHelper.cs:448]: Uses an enum now, looks like this:
image

File: MediaBrowser.Model/Session/HardwareEncodingType.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+{
+    /// <summary>
+    /// Enum HardwareEncodingType.
+    /// </summary>
+    public enum HardwareEncodingType
+    {
+        /// <summary>
+        /// AMD AMF
+        /// </summary>
+        AMF,
```
Comment (crobibero) [inline MediaBrowser.Model/Session/HardwareEncodingType.cs:11]: Please also set the value for all enums
```suggestion
        AMF = 1,
```

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Api/Helpers/TranscodingJobHelper.cs [hunk 5]
```diff
@@ -444,6 +444,10 @@ private void DeleteHlsPartialStreamFiles(string outputFilePath)
             {
                 var audioCodec = state.ActualOutputAudioCodec;
                 var videoCodec = state.ActualOutputVideoCodec;
+                var hardwareAccelerationTypeString = _serverConfigurationManager.GetEncodingOptions().HardwareAccelerationType;
+                HardwareEncodingType? hardwareAccelerationType = string.IsNullOrEmpty(hardwareAccelerationTypeString)
+                    ? null
+                    : (HardwareEncodingType)Enum.Parse(typeof(HardwareEncodingType), hardwareAccelerationTypeString, true);
```
Comment (cvium) [inline Jellyfin.Api/Helpers/TranscodingJobHelper.cs:450]: Enum.Parse is terribly inefficient. Given that the enum is converted to a string in the API we're better off using const strings.

**Review by cvium — APPROVED** [review]
Future work: Use the new enum for `HardwareAccelerationType`

**Pull request merged**